gpt_index/indices/base.py

```diff
@@ -452,6 +457,22 @@ def load_from_disk(cls, save_path: str, **kwargs: Any) -> "BaseGPTIndex":
             file_contents = f.read()
             return cls.load_from_string(file_contents, **kwargs)
 
+    def save_to_dict(self, **save_kwargs: Any) -> dict:
+        """Save to dict."""
+        if self.docstore.contains_index_struct(
+            exclude_ids=[self.index_struct.get_doc_id()]
+        ):
+            raise ValueError(
+                "Cannot call `save_to_string` on index if index is composed on top of "
```

Comment:
nit: not save_to_string in this case (maybe just take that part out)

Reply:
reworded.